include/sys/uio.h

```diff
@@ -17,11 +17,20 @@ ssize_t writev(int fd, const struct iovec *iov, int iovcnt);
 
 typedef struct vm_map vm_map_t;
 
+/* This structure stores just enough information about an uio
+ * to be able to restore it after the uio is modified by uiomove(). */
+typedef struct {
+  size_t us_resid;   /* Saved value of uio_resid */
+  int us_iovcnt;     /* Saved value of uio_iovcnt */
+  size_t us_iov_off; /* Saved value of uio_iov_off */
+} uiosave_t;
```

Comment:
I guess `uiostate_t` is a better name for it cause you want to save & restore `uio` internal state.

Also remember that we define structure types as follows: `typedef struct typename { ... } typename_t`.

Reply:
I chose `uiosave_t` because `uio_t` _is_ the `uio` state. It's not a big deal though, so I have changed it.